src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -570,27 +585,13 @@ HttpResponseOutcome AWSClient::AttemptOneRequest(const std::shared_ptr<Aws::Http
         *m_telemetryProvider->getMeter(this->GetServiceClientName(), {}),
         {{TracingUtils::SMITHY_METHOD_DIMENSION, request.GetServiceRequestName()},{TracingUtils::SMITHY_SERVICE_DIMENSION, this->GetServiceClientName()}});
 
-    if (request.ShouldValidateResponseChecksum())
+    context.SetTransmitResponse(httpResponse);
+    for (const auto& interceptor : m_interceptors)
```

Comment:
what if interceptors is empty. will the workflow still be correct, if not maybe return an error

Reply:
m_interceptors could be empty in which case we apply zero interceptors. infact a optimization we could make in teh future is make per-client interceptors. right now all clients will execute a checksum interceptors as they did before. but in the future we could optimize it where if a client doesnt do any checksumming, it will not compose the interceptor